iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -214,61 +214,222 @@ inline bool string<Capacity>::unsafe_assign(const std::string& str) noexcept
+template <typename T>
+inline
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, std::string>::value),
+                            bool>::type
+    string<Capacity>::operator<(const T& rhs) const noexcept
+{
+    return (compare(rhs) < 0);
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, std::string>::value),
+                            bool>::type
+    string<Capacity>::operator<=(const T& rhs) const noexcept
+{
+    return (compare(rhs) <= 0);
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, std::string>::value),
+                            bool>::type
+    string<Capacity>::operator>(const T& rhs) const noexcept
+{
+    return (compare(rhs) > 0);
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline
+    typename std::enable_if<(is_cxx_string<T>::value || is_char_array<T>::value || std::is_same<T, std::string>::value),
+                            bool>::type
+    string<Capacity>::operator>=(const T& rhs) const noexcept
+{
+    return (compare(rhs) >= 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator==(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) == 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator!=(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) != 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator<(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) > 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator<=(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) >= 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator>(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) < 0);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename std::enable_if<(is_char_array<T>::value || std::is_same<T, std::string>::value), bool>::type
+operator>=(const T& lhs, const string<Capacity>& rhs) noexcept
+{
+    return (rhs.compare(lhs) <= 0);
+}
+
+template <uint64_t Capacity>
+inline int64_t string<Capacity>::compare(const char& other) const noexcept
+{
+    auto result = memcmp(c_str(), &other, 1U);
```

Comment:
Yes, I deliberately didn't add the check for an empty string but I forgot to implement a test for this case and will add it.

First I thought that `memcmp` is overkill, too. But the code is more readable without writing the comparison of one char and in case of equality add the size check.